episodes/superspreading-estimate.Rmd

```diff
@@ -0,0 +1,836 @@
+
+```{r}
+## first, make an epicontacts object
+epi_contacts <-
+  epicontacts::make_epicontacts(
+    linelist = mers_korea_2015$linelist,
+    contacts = mers_korea_2015$contacts
+  )
+```
+
+```{r,eval=FALSE}
+# visualise contact network
+epicontacts::vis_epicontacts(epi_contacts)
+```
+
+```{r,echo=FALSE}
+# visualise contact network
+network <-
+  vis_epicontacts(epi_contacts) %>%
+  visNetwork::visPhysics(solver = "barnesHut")
+
+fname <- "network.html"
+
+visNetwork::visSave(network, fname)
+
+webshot::webshot(
+  fname,
+  delay = 5,
+  zoom = 10
+)
+```
+
+::::::::::::::::::::::::::: spoiler
+
+### Is contact data tidy?
+
+Contact data from a transmission chain can provide information on which infected individuals came into contact with others. We expect to have the infector (`from`) and the infectee (`to`) plus additional columns of variables related to their contact, such as location (`exposure`) and date of contact.
+
+Following [tidy data](https://tidyr.tidyverse.org/articles/tidy-data.html#tidy-data) principles, the observation unit in our contact dataset is the **infector-infectee** pair. Although one infector can infect multiple infectees, from contact tracing investigations we may record contacts linked to more than one infector (e.g. within a household). But we should expect to have unique infector-infectee pairs, because typically each infected person will have acquired the infection from one other.
+
+To ensure these unique pairs, we can check on replicates for infectees:
+
+```{r}
+# no infector-infectee pairs are replicated
+epi_contacts %>%
+  pluck("contacts") %>%
+  group_by(to) %>%
+  filter(n() > 1)
+```
+
+:::::::::::::::::::::::::::
+
+When each infector-infectee row is unique, the number of entries per infector corresponds to the number of secondary cases generated by that individual.
+
+```{r}
+# count secondary cases per infector
+infector_secondary <- epi_contacts %>%
+  pluck("contacts") %>%
+  count(from, name = "secondary_cases")
+```
+
+But this output only contains number of secondary cases for reported infectors, not for each of the individuals in the whole `epicontacts` object.
+
+To get this, first, we can use `epicontacts::get_id()` to get the full list of unique identifiers ("id") from the `epicontacts` class object. Second, join it with the count secondary cases per infector stored in the `infector_secondary` object. Third, replace the missing values with `0` to express no report of secondary cases from them.
+
+```{r,message=FALSE,warning=FALSE}
+all_secondary <- epi_contacts %>%
+  # extract ids in contact *and* linelist using "which" argument
+  epicontacts::get_id(which = "all") %>%
+  # transform vector to dataframe to use left_join()
+  enframe(name = NULL, value = "from") %>%
```

Comment:
This segment contains useful information bout  contacts data, and I it would be good to put in separate section.

Reply:
added as issue https://github.com/epiverse-trace/tutorials-middle/issues/65